grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -152,6 +153,18 @@ class InvoiceService {
         invoiceItem.delete()
     }
 
+    def addItems(Invoice invoice, List items) {
```

Comment:
Let's find a way to do add/update in one service method so we don't need to add an RPC endpoint on the API. I want this to be a clean REST API that we use as the standard to refactor everything else.

Something like this should work.
```
def updateInvoiceItems(Invoice invoice, List items) { 
  items.each { item ->
    InvoiceItem existingInvoiceItem = invoice.invoiceItems.find { it.id == item.id }
    if (existingInvoiceItem) { 
        existingInvoiceItem.quantity = item.quantity
    }
    else { 
      InvoiceCandidate candidateItem = InvoiceCandidate.get(item.id)
      if (candidateItem) {
          InvoiceItem invoiceItem = invoiceItemService.createFromCandidate(candidateItem)
          invoiceItem.quantity = item.quantityToInvoice
          invoice.addToInvoiceItems(invoiceItem)
      }
    }
  }
}
```